airflow-core/newsfragments/48218.significant.rst

```diff
@@ -0,0 +1,14 @@
+Remove option for unlimited parallelism.
+
+Before Airflow 3.0 it was possible to set unlimited parallelism by setting ``[core] parallelism`` to 0. This was removed in Airflow 3.0.0.
+
+* Types of change
+
+  * [ ] Dag changes
+  * [x] Config changes
+  * [ ] API changes
+  * [ ] CLI changes
+  * [x] Behaviour changes
+  * [ ] Plugin changes
+  * [ ] Dependency changes
+  * [ ] Code interface changes
```

Comment:
Do we want to check whether `core.parallelism` is 0? if so, we'll need to add a migration rule in `airflow config lint`. You can leave the implementation part to @sunank200 and me but if we want to add it. We'll need to list it down in the newsfragments

Reply:
Yeah, I hope I can make the config lint addition in a separate PR the next days. But if you are faster than me, this is welcome.
I assume the value should not be auto-migrated (to what?) but an error should be raised.

Reply:
Here we are :-) #48423